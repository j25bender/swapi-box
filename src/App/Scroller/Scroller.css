body {
  overflow: hidden;
}

.scrollText {
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 100vw;
  align-items: center;
  font-size: 75px;
  background-image: url('../../images/star-background.jpg');
  color: whitesmoke;
  perspective: 400px;
}

.crawl {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

@keyframes crawl {
  0% {
    top: 0;
    /* Rotate the text 20 degrees but keep it close to the viewer */
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    /* This is a magic number, but using a big one to make sure the text is fully off the screen at the end */
    top: -6000px;
    /* Slightly increasing the rotation at the end and moving the text far away from the viewer */
    transform: rotateX(25deg) translateZ(-2500px);
  }
}