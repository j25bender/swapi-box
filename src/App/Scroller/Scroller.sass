.scrollText 
  height: 100vh
  overflow: hidden
  display: flex
  flex-direction: column
  align-items: center
  background-image: url('../../images/star-background.jpg')
  color: whitesmoke
  perspective: 500px


.crawl
  height: 60vh
  width: 85vw
  font-size: 50px
  position: relative
  top: -100px
  transform-origin: 50% 100%
  animation: crawl 60s linear infinite


@keyframes crawl 
  0% 
    top: 0
    transform: rotateX(20deg) translateZ(0)
  
  100%  
    top: -6000px
    transform: rotateX(25deg) translateZ(-2500px)
  

