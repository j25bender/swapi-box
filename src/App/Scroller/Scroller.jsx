import React from 'react'
import './Scroller.css'

const Scroller = (props) => {
    if(props.fetch) {
        const randoNum = Math.ceil(Math.random() * Math.ceil(7))
        const randoScroll = props.fetch[randoNum]
        const renderScroll = randoScroll
        return (
            <article className="scrollText">
                <div className="crawl">
                    <h5>{renderScroll}</h5>
                </div>
            </article>
        )

    }
}

export default Scroller