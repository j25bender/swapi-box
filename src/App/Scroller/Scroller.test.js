import React from 'react'
import Enzyme, {shallow, mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Scroller from './Scroller'

Enzyme.configure({ adapter: new Adapter() })

describe('Scroller', () => {
    const mockselectCard = jest.fn()

    it('matches the snapshot', () => {
        const filmTextState = ["It is a dark time for the Rebellion.", "The Empire Strikes Back", "1980-05-17"]

        const wrapper = mount(<Scroller fetch={ filmTextState } />)
        expect(wrapper).toMatchSnapshot()
    })
})