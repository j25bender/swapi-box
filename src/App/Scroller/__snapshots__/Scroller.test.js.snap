// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroller matches the snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Scroller
    fetch={
        Array [
            "It is a dark time for the Rebellion.",
            "The Empire Strikes Back",
            "1980-05-17",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "fetch": Array [
        "It is a dark time for the Rebellion.",
        "The Empire Strikes Back",
        "1980-05-17",
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": <article
        class="scrollText"
>
        <div
                class="crawl"
        >
                <h5 />
        </div>
</article>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
>
          <h5 />
</div>,
        "className": "scrollText",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="crawl"
>
            <h5 />
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h5 />,
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <h5 />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "h5",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "fetch": Array [
          "It is a dark time for the Rebellion.",
          "The Empire Strikes Back",
          "1980-05-17",
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": <article
          class="scrollText"
>
          <div
                    class="crawl"
          >
                    <h5 />
          </div>
</article>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <h5 />
</div>,
          "className": "scrollText",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="crawl"
>
              <h5 />
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h5 />,
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h5 />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "h5",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
